Python/compile.c

```diff
@@ -4918,25 +4948,44 @@ stackdepth_walk(struct compiler *c, basicblock *b, int depth, int maxdepth)
             maxdepth = depth;
         assert(depth >= 0); /* invalid code or bug in stackdepth() */
         if (instr->i_jrel || instr->i_jabs) {
+            /* Recursively inspect jump target */
             target_depth = depth;
+            if (instr->i_opcode == CONTINUE_LOOP) {
+                /* Pops variable number of values from the stack,
+                 * but the target should be already proceeding.
+                 */
+                assert(instr->i_target->b_seen);
+                assert(instr->i_target->b_startdepth <= depth);
+                goto out; /* remaining code is dead */
+            }
             if (instr->i_opcode == FOR_ITER) {
-                target_depth = depth-2;
+                target_depth = depth - 2;
             }
             else if (instr->i_opcode == SETUP_FINALLY ||
-                     instr->i_opcode == SETUP_EXCEPT) {
-                target_depth = depth+3;
-                if (target_depth > maxdepth)
-                    maxdepth = target_depth;
+                     instr->i_opcode == SETUP_EXCEPT)
+            {
+                depth = depth - 6;
+            }
+            else if (instr->i_opcode == SETUP_WITH ||
+                     instr->i_opcode == SETUP_ASYNC_WITH)
+            {
+                depth = depth - 5;
```

Comment:
Same here.

Reply:
Same as above. See comments in PyCompile_OpcodeStackEffect().